agents/ahv/fence_ahv.py

```diff
@@ -0,0 +1,608 @@
+
+    if not vmid and not name:
+        logging.error("Need VM name or VM UUID for power op")
+        fail(EC_GENERIC_ERROR)
+    try:
+        power_state = client.get_power_state(vm_name=name, vm_uuid=vmid)
+    except AHVFenceAgentException:
+        fail(EC_GENERIC_ERROR)
+
+    return power_state
+
+def set_power_status(client, options):
+    vmid = None
+    name = None
+    action = None
+    timeout = None
+
+    if "--action" not in options:
+        logging.error("No power op action specified")
+        fail(EC_GENERIC_ERROR)
+
+    action = options["--action"].lower()
+
+    if "--power-timeout" in options:
+        timeout = options["--power-timeout"]
+
+    if "--uuid" in options:
+        vmid = options["--uuid"]
+    else:
+        name = options["--plug"]
+
+    if not name and not vmid:
+        logging.error("Need VM name or VM UUID to set power state of a VM")
+        fail(EC_GENERIC_ERROR)
+
+    try:
+        client.set_power_state(vm_name=name, vm_uuid=vmid,
+                               power_state=action, timeout=timeout)
+    except AHVFenceAgentException as err:
+        logging.error(err)
+        fail(EC_GENERIC_ERROR)
+    except TaskTimedOutException as err:
+        logging.error(err)
+        fail(EC_TIMED_OUT)
+
+def power_cycle(client, options):
+    vmid = None
+    name = None
+    timeout = None
+
+    if "--power-timeout" in options:
+        timeout = options["--power-timeout"]
+
+    if "--uuid" in options:
+        vmid = options["--uuid"]
+    else:
+        name = options["--plug"]
+
+    if not name and not vmid:
+        logging.error("Need VM name or VM UUID to set power cycling a VM")
+        fail(EC_GENERIC_ERROR)
+
+    try:
+        client.power_cycle_vm(vm_name=name, vm_uuid=vmid, timeout=timeout)
+    except AHVFenceAgentException as err:
+        logging.error(err)
+        fail(EC_GENERIC_ERROR)
+    except TaskTimedOutException as err:
+        logging.error(err)
+        fail(EC_TIMED_OUT)
+
+def define_new_opts():
+    all_opt["filter"] = {
+            "getopt": ":",
+            "longopt": "filter",
+            "help": """
+            --filter=[filter]	Filter list, list VMs actions.
+            --filter=\"name eq 'node1-vm'\"
+            --filter=\"startswith(name,'node')\"
+            --filter=\"name in ('node1-vm','node-3-vm')\" """,
+            "required": "0",
+            "shortdesc": "Filter list, get_list"
+            "e.g: \"name eq 'node1-vm'\"",
+            "order": 2
+            }
+
+def main():
+    device_opt = [
+            "ipaddr",
+            "login",
+            "passwd",
+            "ssl",
+            "notls",
+            "web",
+            "port",
+            "filter",
+            "method",
+            "disable_timeout",
+            "power_timeout"
+            ]
+
+    atexit.register(atexit_handler)
+    define_new_opts()
+
+    all_opt["method"]["default"] = "onoff"
+    all_opt["disable_timeout"]["default"] = "false"
```

Comment:
This shouldnt be changed, as Pacemaker should do the timeout. The parameter is there to be able to use the built-in timeouts when run from CLI.

Reply:
Fixed.